src/modules/fossil_metrics.rs

```diff
@@ -0,0 +1,275 @@
+
+/// Creates a module with currently added/deleted lines in the Fossil check-out in the current
+/// directory.
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("fossil_metrics");
+    let config = FossilMetricsConfig::try_load(module.config);
+
+    // As we default to disabled=true, we have to check here after loading our config module,
+    // before it was only checking against whatever is in the config starship.toml
+    if config.disabled {
+        return None;
+    };
+
+    if !is_checkout(context) {
+        return None;
+    }
+
+    // Read the total number of added and deleted lines from "fossil diff --numstat"
+    let output = context.exec_cmd("fossil", &["diff", "--numstat"])?.stdout;
+    let stats = FossilDiff::parse(&output, config.only_nonzero_diffs);
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        formatter
+            .map_style(|variable| match variable {
+                "added_style" => Some(Ok(config.added_style)),
+                "deleted_style" => Some(Ok(config.deleted_style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "added" => Some(Ok(stats.added)),
+                "deleted" => Some(Ok(stats.deleted)),
+                _ => None,
+            })
+            .parse(None, Some(context))
+    });
+
+    module.set_segments(match parsed {
+        Ok(segments) => segments,
+        Err(error) => {
+            log::warn!("Error in module `fossil_metrics`:\n{}", error);
+            return None;
+        }
+    });
+
+    Some(module)
+}
+
+fn is_checkout(ctx: &Context) -> bool {
+    let dev_id = ctx.current_dir.device_id();
+    for dir in ctx.current_dir.as_path().ancestors() {
+        if dev_id != dir.device_id() {
+            break;
+        }
+        if dir.join(".fslckout").is_file() {
+            return true;
+        }
+    }
+    false
+}
+
+/// Represents the parsed output from a Fossil diff with the --numstat option enabled.
+#[derive(Debug, PartialEq)]
+struct FossilDiff<'a> {
+    added: &'a str,
+    deleted: &'a str,
+}
+
+impl<'a> FossilDiff<'a> {
+    /// Parses the output of `fossil diff --numstat` as a `FossilDiff` struct.
+    pub fn parse(diff_numstat: &'a str, only_nonzero_diffs: bool) -> Self {
+        // Fossil formats the last line of the output as "%10d %10d TOTAL over %d changed files\n"
+        // where the 1st and 2nd placeholders are the number of added and deleted lines respectively
+        let total_line = diff_numstat.lines().last().unwrap_or("");
+        let mut split = total_line.split_whitespace();
+
+        let zero = if only_nonzero_diffs { "" } else { "0" };
+
+        let added = match split.next() {
+            Some("0") | None => zero,
+            Some(s) => s,
+        };
+
+        let deleted = match split.next() {
+            Some("0") | None => zero,
+            Some(s) => s,
+        };
```

Comment:
Since this operation isn't supported on some versions, perhaps try sanity-check the returned values.

Reply:
Solved by using a regex to only match on the expected format.